src/WebJobs.Extensions.DurableTask/Correlation/TelemetryActivator.cs

```diff
@@ -93,7 +94,14 @@ private void SetUpTelemetryClient()
             telemetryInitializer.ExcludeComponentCorrelationHttpHeadersOnDomains.Add("127.0.0.1");
             config.TelemetryInitializers.Add(telemetryInitializer);
 
-            config.InstrumentationKey = Environment.GetEnvironmentVariable("APPINSIGHTS_INSTRUMENTATIONKEY");
+            if (Environment.GetEnvironmentVariable("APPINSIGHTS_INSTRUMENTATIONKEY") != null)
+            {
+                config.InstrumentationKey = Environment.GetEnvironmentVariable("APPINSIGHTS_INSTRUMENTATIONKEY");
+            }
+            else
+            {
+                logger.LogWarning("'APPINSIGHTS_INSTRUMENTATIONKEY' isn't defined in the current environment variables, but Distributed Tracing is enabled. Please set 'APPINSIGHTS_INSTRUMENTATIONKEY' to use Distributed Tracing.");
```

Comment:
I wonder if it would be better to use `EndToEndTraceHelper.ExtensionWarningEvent` instead of `ILogger.LogWarning`. That way we can also see this warning in Kusto. The problem, however, is that `EndToEndTraceHelper` is an internal class and we can't simply pass it to public methods on public classes. If we agree that `EndToEndTraceHelper` is a better solution for logging, then maybe we can make `Initialize()` internal, or have a separate internal `Initialize()` method if it's not practical to change the existing one?

Reply:
It would be nice to use `EndToEndTraceHelper` so we have this information in Kusto. If we go with this approach then making `Initialize()` internal doesn't look like it would work. I tried it out and an error showed up saying that internal isn't a valid access modifier for `Initialize()` in C# 7.3. It's allowed in C# 8.0 though.

Reply:
This problem comes up pretty frequently with types that use our dependency injection framework. Essentially, the Functions DI model requires our constructor is public, which means all the interfaces (and all of their parameters) need to be public. Options:

1. Don't use DI for instantiating ITelemetryActivator, and just instantiate it in the constructor directly. We may lose some testability benefits here, but it looks like we don't allow customers to customize this via DI anyways, so we wouldn't lose those advantages.
2. EndToEndTraceHelper essentially just takes a logger, so you could just reinstantiate one in your `Initialize(ILogger logger)` method directly.
3. We try and come up with a more general solution for this problem. We could see if Functions V2/V3 allows us to provide a constructor that just takes the IServiceCollection, and we pull the services we want from that IServiceCollection in the constructor implementation. This would prevent us from having to make all of the services we inject public, but I'm not sure if this works without some modifications to the Functions/WebJobs layer...

Reply:
@ConnorMcMahon thanks for listing out some different options! With option 1, we probably don't want to take away from any testing benefits. Option 2 might be the simplest option since it would reinstantiate `EndToEndTraceHelper` in the scope of `TelemetryActivator`. Is there an option that you are leaning more towards using?

Reply:
I realize that thumbs up didn't answer your question. I think option 2 is the easiest approach to get to where we need.

Reply:
Sounds good, I'll implement option 2.